### PR TITLE
Method deepst jieran 

### LINKED ISSUE #22: method_DeepST
**Update metadata:**
https://docs.google.com/spreadsheets/d/1QCeAF4yQG4bhZSGPQwwVBj_XF7ADY_2mK5xivAIfHsc/edit#gid=0

**Details of method (publication, GitHub):**
https://academic.oup.com/nar/article/50/22/e131/6761985
https://github.com/JiangBioLab/DeepST

### PR DESCRIPTION
**Pull request by Jieran-S**
Link to #22 

Some issues I found during the implementation:

1. Based on our framework, the data augmentation process, based on their pre-trained DNN on the H&E image cannot be implemented, it could be possible that the performance can be compromised. But we can add this augmentation as an option in our preprocessing phase.
2. The model is a DNN model and it takes quite a long time to run even for our small test dataset (4226 × 2593) the algorithm takes over 15 (config_2) and 8 (config_1) minutes to train, that's without the data augmentation step mentioned above.
3. This method cannot assign seed or any random state...according to [their issue](https://github.com/JiangBioLab/DeepST/issues/6), their results cannot be reproduced even with the same configuration and data.

### CODE REVIEW

File: method/DeepST/DeepST.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env python
+
+# Author_and_contribution: Niklas Mueller-Boetticher; created template
+# Author_and_contribution: ENTER YOUR NAME AND CONTRIBUTION HERE
```
Comment (niklasmueboe) [inline method/DeepST/DeepST.py:4]: Enter conribution

File: method/DeepST/DeepST.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+import argparse
+
+# TODO adjust description
+parser = argparse.ArgumentParser(description="Method ...")
```
Comment (niklasmueboe) [inline method/DeepST/DeepST.py:9]: just write maybe a oneliner

File: method/DeepST/DeepST.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    else:
+        adata.uns["image"] = None
+
+    return adata
+
+adata = get_anndata(args)
+adata.var_names_make_unique()
+
+#######
+# NOTE: Seed cannot be used here
```
Comment (niklasmueboe) [inline method/DeepST/DeepST.py:138]: Wondering if we should still try to seed

https://pytorch.org/docs/stable/notes/randomness.html

File: method/DeepST/DeepST.py [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+                 "adj_norm": adj_norm,
+                 "adj_label": adj_label,
+                 "norm_value": norm }
+
+    # NOTE: There're so many hyperparameters but I would not change any...the last few commits the author made is to change some default parameters so I assume the author has figure out the "best" config for this model
+    
+    # Conv_type: ["GCNConv", "SAGEConv", "GraphConv", "GatedGraphConv", "ResGatedGraphConv", "TransformerConv", "TAGConv", "ARMAConv", "SGConv", "MFConv", "RGCNConv", "FeaStConv", "LEConv", "ClusterGCNConv"]
+    data = adata.obsm["reduced_dimensions"].astype(np.float64)
+    
+    deepst_embed = deepen._fit(data,
```
Comment (niklasmueboe) [inline method/DeepST/DeepST.py:208]: Is this part of the public API

**Review by niklasmueboe — CHANGES_REQUESTED** [review]

File: method/DeepST/DeepST.py (hunk 4 above)
Comment (Jieran-S) [reply method/DeepST/DeepST.py:208]: Do you mean the conv_type? They don't have ANY documentation on their model but in[ their script ](https://github.com/JiangBioLab/DeepST/blob/main/deepst/model.py#L63)they gave those options...

File: method/DeepST/DeepST.py (hunk 2 above)
Comment (Jieran-S) [reply method/DeepST/DeepST.py:9]: I want to give it the same amount of description they gave on their code

File: method/DeepST/DeepST.py (hunk 4 above)
Comment (niklasmueboe) [reply method/DeepST/DeepST.py:213]: No I meant the `._fit`, usually underscore would indicate private functions, but it seems their whole API is built like that

Comment (Jieran-S) [reply method/DeepST/DeepST.py:213]: Oh yes all the main functions are starting with a “_”. It should be public as they also use those functions in their github page example

**Review by niklasmueboe — APPROVED** [review]
